lib/src/over_react_test/custom_matchers.dart

```diff
@@ -337,3 +339,139 @@ Matcher throwsPropError_Combination(String propName, String prop2Name, [String m
+    } else if (x is String) {
+      return anyElement(contains(x));
+    } else {
+      return equals(x);
+    }
+  }
+
+  @override
+  featureValueOf(actual) {
+    var logs = <String>[];
+
+    if (actual is List) return actual;
+
+    if (actual is! Function()) {
+      throw ArgumentError('The actual value must be a callback or a List.');
+    }
+
+    try {
+      logs = recordConsoleLogs(actual, config ?? logConfig);
+    } catch (_) {}
+
+    return logs;
+  }
+}
+
+/// A Matcher used to compare a list of logs against a provided matcher.
+///
+/// __Examples:__
+///
+///     When passed a string, the matcher look for any log that contains that
+///     substring.
+///     ```dart
+///       expect(callbackFunction, emitsLogs('I expect this log'));
+///     ```
+///
+///     When passed a list, the matcher will do an equality check on the actual
+///     log List.
+///
+///     Alternatively, the `String` can be wrapped in a `contains` to check the
+///     if the comparable index contains that substring.
+///     ```dart
+///       expect(callbackFunction, emitsLogs(['I expect this log', 'And this Log']));
+///       expect(callbackFunction, emitsLogs([
+///         contains('I expect this log'),
+///         contains('And this Log'),
+///       ]));
+///     ```
+///
+///     All usual `Iterable` matchers can also be used.
+///     ```dart
+///       expect(callbackFunction, emitsLogs(containsAll(['I expect this log'])));
+///       expect(callbackFunction, emitsLogs(containsAllInOrder(['I expect this log'])));
+///       expect(callbackFunction, emitsLogs(hasLength(1)));
+///     ```
+///
+/// Related: [emitsNoLogs]
+Matcher emitsLogs(dynamic expected, {ConsoleConfiguration consoleConfig}) =>
+    _LoggingFunctionMatcher(expected, config: consoleConfig);
+
+/// A matcher to verify that a callback function does not emit any logs.
+///
+/// Related: [emitsLogs]
+final Matcher emitsNoLogs = _LoggingFunctionMatcher(isEmpty);
+
+/// The string used to identify a `propType` error.
+const _propTypeErrorMessage = 'Failed prop type';
+
+/// A matcher used to assert expected a `List` contains expected `propType`
+/// warnings.
+///
+/// The actual value can be a callback function, which will result in the matcher
+/// asserting that the expected `propType` warnings appear during the runtime
+/// of the callback.
+///
+/// Related: [_LoggingFunctionMatcher]
+class _PropTypeLogMatcher extends _LoggingFunctionMatcher {
+  _PropTypeLogMatcher(expected)
+      : super(expected,
+      description: 'emits the propType warning',
+      name: 'propType warning');
+
+  final _filter = contains(_propTypeErrorMessage);
+
+  @override
+  featureValueOf(actual) {
+    if (actual is! Function() && actual is! List) {
+      throw ArgumentError('The actual value must be a callback or a List.');
+    }
+
+    try {
+      actual = actual is List ? actual : recordConsoleLogs(actual, errorConfig);
+      actual.removeWhere((log) => !_filter.matches(log, {}));
```

Comment:
This is a dynamic call, and we probably shouldn't be mutating the `actual` being passed in. Could we add some typing and make a copy?

```dart
    var logs = actual is List ? actual : recordConsoleLogs(actual, errorConfig);
    return logs.where((log) => _filter.matches(log, {})).toList();
```